internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,687 @@
+	}
+
+	if input.Name != "" {
+		output.Name = &input.Name
+	}
+
+	return output
+}
+
+func expandPlatformConfigurationModel(inputList []PlatformConfigurationModel) (packetcorecontrolplane.PlatformConfiguration, error) {
+	output := packetcorecontrolplane.PlatformConfiguration{}
+	if len(inputList) == 0 {
+		return output, nil
+	}
+
+	input := inputList[0]
+
+	output.Type = packetcorecontrolplane.PlatformType(input.Type)
+
+	if pass, err := vertifyPlatformConfigurationModel(input); !pass {
+		return output, err
+	}
+
+	if input.AzureStackEdgeDeviceId != "" {
+		output.AzureStackEdgeDevice = &packetcorecontrolplane.AzureStackEdgeDeviceResourceId{
+			Id: input.AzureStackEdgeDeviceId,
+		}
+	}
+
+	if input.ConnectedClusterId != "" {
+		output.ConnectedCluster = &packetcorecontrolplane.ConnectedClusterResourceId{
+			Id: input.ConnectedClusterId,
+		}
+	}
+
+	if input.AzureStackHciClusterId != "" {
+		output.AzureStackHciCluster = &packetcorecontrolplane.AzureStackHCIClusterResourceId{
+			Id: input.AzureStackHciClusterId,
+		}
+	}
+
+	if input.CustomLocationId != "" {
+		output.CustomLocation = &packetcorecontrolplane.CustomLocationResourceId{
+			Id: input.CustomLocationId,
+		}
+	}
+
+	return output, nil
+}
+
+func vertifyPlatformConfigurationModel(input PlatformConfigurationModel) (bool, error) {
+	idList := make([]string, 0)
+	if input.AzureStackEdgeDeviceId != "" {
+		idList = append(idList, input.AzureStackEdgeDeviceId)
+	}
+	if input.AzureStackHciClusterId != "" {
+		idList = append(idList, input.AzureStackHciClusterId)
+	}
+	if input.ConnectedClusterId != "" {
+		idList = append(idList, input.ConnectedClusterId)
+	}
+
+	if len(idList) == 0 {
+		return false, fmt.Errorf("at least one of `azure_arc_connected_cluster_id`, `azure_stack_hci_cluster_id` and `custom_location_id` should be specified")
+	}
+
+	firstId := idList[0]
+	for _, id := range idList {
+		if !strings.EqualFold(firstId, id) {
+			return false, fmt.Errorf("if multiple of `azure_arc_connected_cluster_id`, `azure_stack_hci_cluster_id` and `custom_location_id` are specified, they should be consistent with each other")
+		}
+	}
+
+	return true, nil
+}
+
+func flattenPacketCoreControlPlaneInterfacePropertiesModel(input packetcorecontrolplane.InterfaceProperties) []InterfacePropertiesModel {
+	var outputList []InterfacePropertiesModel
+
+	output := InterfacePropertiesModel{}
+
+	if input.IPv4Address != nil {
+		output.IPv4Address = *input.IPv4Address
+	}
+
+	if input.IPv4Gateway != nil {
+		output.IPv4Gateway = *input.IPv4Gateway
+	}
+
+	if input.IPv4Subnet != nil {
+		output.IPv4Subnet = *input.IPv4Subnet
+	}
+
+	if input.Name != nil {
+		output.Name = *input.Name
+	}
+
+	outputList = append(outputList, output)
+	return outputList
+}
+
+func flattenPlatformConfigurationModel(input packetcorecontrolplane.PlatformConfiguration) []PlatformConfigurationModel {
+	var outputList []PlatformConfigurationModel
+
+	output := PlatformConfigurationModel{
+		Type: string(input.Type),
```

Comment:
(as above) we generally don't expose the `type` field, what's the value for doing so here?

Reply:
it's used to indentify the type of platform, just a parameter pass to service..

Reply:
AFAIK, the `type` of `platform` could not be told by the input of other parameters. AFAIK, the only way is to specify it manually.